src/views/TimelineTextView.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import {ChangeEvent, useCallback} from "react";
+import {MarkerFileData} from "../dataTypes";
+
+interface Props {
+    frameStart: number,
+    frame: number, setFrame: React.Dispatch<React.SetStateAction<number>>,
+    frameEnd: number,
+
+    frameCropStart: number, setCropStart: React.Dispatch<React.SetStateAction<number>>,
+    frameCropEnd: number, setCropEnd: React.Dispatch<React.SetStateAction<number>>,
+
+    frameRef: React.MutableRefObject<number>,
```

Comment:
I'd rather not pass this in, you can create a ref in here if you need one

Reply:
which I don't think you do in this case, the function version of the state setter should cover these needs

Reply:
Similar to controlsHelpImgNum above, I overlooked that this could be moved in to the view where it's used. Fixed.

I chose to keep the ref around as it was originally used in togglePlaying:
```typescript
const togglePlaying = useCallback(() => {
	setPlaying(current => {
		if(current) return false;
		if(frameRef.current >= markerFileData.frames.length - 1) setFrame(frameStart); // restart if at end
		return true;
	});
}, [markerFileData.frames.length, frameRef, frameStart]);
```

I figured we wouldn't want any callback that would otherwise depend on the frame state variable to memoize for every new value of frame.